packages/flutter/lib/src/material/tabs.dart

```diff
@@ -591,6 +665,37 @@ class _TabBarState extends State<TabBar> {
   _IndicatorPainter _indicatorPainter;
   int _currentIndex;
   double _tabStripWidth;
+  List<GlobalKey> _tabKeys;
+
+  @override
+  void initState() {
+    super.initState();
+    _tabKeys = widget.tabs.map((Widget tab) => new GlobalKey()).toList();
+  }
+
+  Decoration get _indicator {
+    if (widget.indicator != null)
+      return widget.indicator;
+
+    Color color = widget.indicatorColor ?? Theme.of(context).indicatorColor;
+    // ThemeData tries to avoid this by having indicatorColor avoid being the
+    // primaryColor. However, it's possible that the tab bar is on a
+    // Material that isn't the primaryColor. In that case, if the indicator
+    // color ends up matching the material's color, then this overrides it.
+    // When that happens, automatic transitions of the theme will likely look
+    // ugly as the indicator color suddenly snaps to white at one end, but it's
+    // not clear how to avoid that any further.
+    if (color == Material.of(context).color)
```

Comment:
Maybe just not do anything? I would think it's the developer's responsibility to not use same background/foreground colors. I don't think we can just decide he is making a mistake (maybe he wants the same color), especially as we don't have a good resolution here. (Also, what happens if the material color is white?)

If we leave this here, we should probably document this behavior in the relevant places  (TabBar.indicatorColor and Theme.indicatorColor)

Reply:
This is old code that I left alone for the sake of backwards compatibility. I'd be happy to document or revise it in a separate PR